### PR TITLE
Display leaf labels in timetree

### PR DESCRIPTION
**Pull request by rlskoeser**

### CODE REVIEW

**Comment by gissoo** [comment]
Thanks, @rlskoeser !! These have been super helpful!

**Review by gissoo — APPROVED** [review]

File: assets/js/timetree.js [hunk 1]
```diff
@@ -8,119 +8,136 @@ import {schemeGreens} from "d3-scale-chromatic"
 const d3 = {select, forceSimulation, forceManyBody, forceCenter, forceCollide,
     forceLink, forceY, line, curveNatural, scaleSequential, schemeGreens};
 
+// load & parse leaf data from json embedded in the document
+const leafData = document.querySelector('.leaf-data');
+const data = JSON.parse(leafData.value);
+
```
Comment (quadrismegistus) [inline assets/js/timetree.js:14]: So we're not using jquery in general?

File: assets/js/timetree.js [hunk 2]
```diff
@@ -8,119 +8,136 @@ import {schemeGreens} from "d3-scale-chromatic"
...
+// generate list of centuries referenced in the data; sort most recent first
+let centuries = Array.from(new Set(
+  data.leaves.filter(leaf => leaf.century != undefined)
+      .map(leaf => leaf.century))
+  ).sort().reverse();
```
Comment (quadrismegistus) [inline assets/js/timetree.js:19]: does precommit format this? just curious. kind of slick

File: assets/js/timetree.js [hunk 3]
```diff
@@ -8,119 +8,136 @@ import {schemeGreens} from "d3-scale-chromatic"
...
+    } else if (leaf.sort_date == "TBA" || leaf.sort_date == "?") {
+      // put TBs / ? in the 2000s
+      leaf.century = 20;
+    }
+  }
+});
 
-fetch('index.json')
-  .then((response) => response.json())
-  .then((data) => {
```
Comment (quadrismegistus) [inline assets/js/timetree.js:14]: Nice. I prefer the new solution where all of this isn't embedded in the fetch function

File: assets/js/timetree.js [hunk 4]
```diff
@@ -8,119 +8,136 @@ import {schemeGreens} from "d3-scale-chromatic"
...
+let centuriesOldestFirst = Array.from(centuries).reverse();
+for (let branch in branches) {
+  centuriesOldestFirst.forEach((c, index) => {
+    let branchId = branch + c;
+    nodes.push({
+      id: branchId,
+      title: branch + " c" +c,
+      type: "branch",
+      century: c,
+    });
```
Comment (quadrismegistus) [inline assets/js/timetree.js:87]: Do these pushes auto-regen the viz? Hope not

File: assets/js/timetree.js [hunk 5]
```diff
@@ -213,11 +233,11 @@ function TreeGraph({nodes, links, centuries}) {
     // .alphaDecay(0.2)
     .force("collide", d3.forceCollide().radius(18))
     // NOTE: may want to adjust to make variable by node type
-    .force("link", d3.forceLink(links))
+    .force("link", d3.forceLink(links).strength(link => 0.8))
```
Comment (quadrismegistus) [inline assets/js/timetree.js:236]: Does this need to be parameterized/set elsewhere?

File: layouts/partials/leaf_data.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+			{{ $pageInfo = merge $pageInfo (dict "branch" . ) }}
+		{{ end }}
+
+		{{ $data.Add "sectionOutput" $pageInfo }}
+	{{ end }}
+
+	{{ $data.SetInMap "output" (lower .CurrentSection.Title) ($data.Get "sectionOutput") }}
+{{ end }}
+
+{{ $data.Get "output" | jsonify (dict "indent" "  ") }}
```
[the remark this answers is not in the document]
Comment (rlskoeser) [reply layouts/partials/leaf_data.json:30]: I moved some of the logic for annotating/organizing the data from javascript into hugo for efficiency (although maybe it's not great for complexity/understanding). Should I add more comments to the template?

You can see the structure it generates here: https://lenape-timetree-dev-pr-33.onrender.com/index.json

File: assets/js/timetree.js (hunk 5 above)
Comment (rlskoeser) [reply assets/js/timetree.js:236]: good thought, maybe — right now I'm just playing with the forces a bit to see how much I can use them to tweak the layout & placement.  I'm currently thinking I'll need to adjust the century pseudo-nodes anyway (i.e., add more nodes for branch+century if there are more nodes), which may negate the need for this adjustment; the default link strength is based on degree, which isn't great for our current leaf+node structure.

File: assets/js/timetree.js (hunk 1 above)
Comment (rlskoeser) [reply assets/js/timetree.js:14]: right — built in js functionality is sufficiently advanced we can use it directly, we don't need the bloat/complication of jquery

File: assets/js/timetree.js (hunk 2 above)
Comment (rlskoeser) [reply assets/js/timetree.js:19]: ooh thanks for reminder, I haven't setup precommit formatting yet but I really should

**Pull request merged**